app/models/concerns/product_stock.rb

```diff
@@ -0,0 +1,25 @@
+require 'active_support/concern'
+
+module ProductStock
+  extend ActiveSupport::Concern
+
+  def on_demand
+    if has_variants?
+      if variants.size > 1
+        raise 'Cannot determine product on_demand value of product with multiple variants'
```

Comment:
what about implementing this as an early return? I think this better conveys the fact that it's an invalid case and it'd reduce a nesting level.

Reply:
do you mean:
```ruby
 if has_variants?
   raise 'Cannot determine product on_demand value of product with multiple variants' if variants.size > 1
   variants.first.on_demand
else
   master.on_demand
end
```